signer/src/block_observer.rs

```diff
@@ -514,9 +514,13 @@ impl<C: Context, B> BlockObserver<C, B> {
                 }
             }
 
-            for output in tx_info.to_outputs(&signer_script_pubkeys) {
+            let (tx_outputs, withdrawal_outputs) = tx_info.to_outputs(&signer_script_pubkeys)?;
```

Comment:
mmm this is not right, I'll change it to always `write_tx_output` and report errors for  `withdrawal_outputs`